core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -259,7 +248,7 @@ private[spark] class TaskSetManager(
     while (indexOffset > 0) {
       indexOffset -= 1
       val index = list(indexOffset)
-      if (!executorIsBlacklisted(execId, index)) {
+      if (!blacklistTracker.fold(false)(_.executorIsBlacklisted(execId, sched))) {
```

Comment:
why we need fold here?  it seems tracker.isBlacklisted(execId, sched) == true/false is enough, right?